### PR TITLE
Starfield

### PR DESCRIPTION
**Pull request by Silarn**
* Add stubs / defaults for overridePluginsAreSupported and secondaryDataDIrectories
* Add zlib decompression support to save parsing

### CODE REVIEW

File: src/gamebryo/gamegamebryo.cpp [hunk 1]
```diff
@@ -69,6 +69,11 @@ QDir GameGamebryo::dataDirectory() const
   return gameDirectory().absoluteFilePath("data");
 }
 
+QMap<QString, QDir> GameGamebryo::secondaryDataDirectories() const
```
Comment (Al12rs) [inline src/gamebryo/gamegamebryo.cpp:72]: I'm not too happy with this solution. I'll probably expand on this in uibase pr.

**Review by Al12rs — COMMENTED** [review]
I don't like the secondary data solution as it seems like a hack specific for starfield and not really usable for any other game.
Perhaps there is a solution that is more flexible and gives more control to game plugins.

Other than that it looked good to me. Didn't look too deep into the saves code.

**Comment by Holt59** [comment]
I trust you on the changes but code should be formatted.

**Review by Holt59 — CHANGES_REQUESTED** [review]
I trust you on the changes but code should be formatted.

**Comment by Silarn** [comment]
I've reformatted the code but I think I need to make one small change to the zlib inflate code. I don't think I need to bake the start location read / seek into that function. I thought I might need the final unpack size but I don't think that's actually necessary. The other thing is I'm still not convinced it's unpacked the entire save. Though we have the data we need ultimately.

**Comment by Silarn** [comment]
My current guess is that I can add up the total processed bytes for the input stream to locate where the next zlib chunk begins. The test extractions I made were always suspiciously 256k long so I have a feeling they broke up the data into 256k chunks.

**Review by Al12rs — APPROVED** [review]

**Pull request merged**